### PR TITLE
DOMA-1125 Fix bug with not editable map in properties

### PR DESCRIPTION
**Pull request by dkoviazin**
1. To reproduce the bug - Start to add a section, don't press the Add Section button, change the name of preview section, press Save property button
2. Added tests

### CODE REVIEW

**Review by sitozzz — APPROVED** [review]

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/property/constants/property.ts [hunk 1]
```diff
@@ -9,6 +9,54 @@ const buildingEmptyMapJson = {
     'sections': [],
 }
 
+const autoFixBuildingMapJson = {
```
Comment (Dimitreee) [inline apps/condo/domains/property/constants/property.ts:12]: add comment

File: apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts [hunk 2]
```diff
@@ -227,16 +247,16 @@ describe('Map constructor', () => {
         describe('Check that JSON schema validator is working', () => {
             it('should react to bad structure', () => {
                 const Building = new MapEdit(null, () => null )
-                Building.map = cloneDeep(notValidBuildingMapJson)
+                Building.map = cloneDeep(notValidBuildingMapJson) as BuildingMap
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts:250]: why typecast?

File: apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts [hunk 3]
```diff
@@ -227,16 +247,16 @@ describe('Map constructor', () => {
...
                 const isValid = Building.validateSchema()
                 expect(isValid).toBe(false)
                 expect(Building.validationErrors).toHaveLength(1)
             })
             it('should pass validation on good structure', () => {
                 const Building = new MapEdit(null, () => null )
-                Building.map = cloneDeep(buildingMapJson)
+                Building.map = cloneDeep(buildingMapJson) as BuildingMap
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts:257]: why typecast?

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts:250]: TS issues warning after we change type from JSON to Apollo schema

File: apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts:257]: TS issues warning after we change type from JSON to Apollo schema

File: apps/condo/domains/property/constants/property.ts (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/property/constants/property.ts:15]: added

**Pull request merged**